src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -210,4 +239,60 @@ private static string GetUserAgentString()
             return UserAgentProduct;
         }
     }
+
+    private void ApplyConfiguration(
+        IConfiguration configuration,
+        OtlpExporterOptionsConfigurationType configurationType)
+    {
+        Debug.Assert(configuration != null, "configuration was null");
+
+        // Note: When using the "AddOtlpExporter" extensions configurationType
+        // never has a value other than "Default" because OtlpExporterOptions is
+        // shared by all signals and there is no way to differentiate which
+        // signal is being constructed.
+        if (configurationType == OtlpExporterOptionsConfigurationType.Default)
+        {
+            this.ApplyConfigurationUsingSpecificationEnvVars(
+                configuration!,
+                OtlpSpecConfigDefinitions.DefaultEndpointEnvVarName,
+                appendSignalPathToEndpoint: true,
+                OtlpSpecConfigDefinitions.DefaultProtocolEnvVarName,
+                OtlpSpecConfigDefinitions.DefaultHeadersEnvVarName,
+                OtlpSpecConfigDefinitions.DefaultTimeoutEnvVarName);
+        }
+        else if (configurationType == OtlpExporterOptionsConfigurationType.Logs)
+        {
+            this.ApplyConfigurationUsingSpecificationEnvVars(
+                configuration!,
+                OtlpSpecConfigDefinitions.LogsEndpointEnvVarName,
+                appendSignalPathToEndpoint: false,
+                OtlpSpecConfigDefinitions.LogsProtocolEnvVarName,
+                OtlpSpecConfigDefinitions.LogsHeadersEnvVarName,
+                OtlpSpecConfigDefinitions.LogsTimeoutEnvVarName);
+        }
+        else if (configurationType == OtlpExporterOptionsConfigurationType.Metrics)
+        {
+            this.ApplyConfigurationUsingSpecificationEnvVars(
+                configuration!,
+                OtlpSpecConfigDefinitions.MetricsEndpointEnvVarName,
+                appendSignalPathToEndpoint: false,
+                OtlpSpecConfigDefinitions.MetricsProtocolEnvVarName,
+                OtlpSpecConfigDefinitions.MetricsHeadersEnvVarName,
+                OtlpSpecConfigDefinitions.MetricsTimeoutEnvVarName);
+        }
+        else if (configurationType == OtlpExporterOptionsConfigurationType.Traces)
+        {
+            this.ApplyConfigurationUsingSpecificationEnvVars(
+                configuration!,
+                OtlpSpecConfigDefinitions.TracesEndpointEnvVarName,
+                appendSignalPathToEndpoint: false,
+                OtlpSpecConfigDefinitions.TracesProtocolEnvVarName,
+                OtlpSpecConfigDefinitions.TracesHeadersEnvVarName,
+                OtlpSpecConfigDefinitions.TracesTimeoutEnvVarName);
+        }
+        else
```

Comment:
Remove this code if we don't expect it to called?

Reply:
It is a defensive thing. This code is here to alert a future dev who might add a new definition to `OtlpExporterOptionsConfigurationType` that they need to update this code in order for things to work.